Helper/Order.php

```diff
@@ -1037,8 +1031,8 @@ public function dispatchPostCheckoutEvents($order, $quote)
             ]
         );
 
-        // Nullify bolt_reserved_order_id. Prevents dispatching more then once.
-        $quote->setBoltReservedOrderId(null);
+        // Set dispatched to be true. Prevents dispatching more then once.
+        $quote->setBoltDispatched(true);
```

Comment:
This wont work among subsequent hooks because it is not persistent anywhere. The old "flag" was added to the table here https://github.com/BoltApp/bolt-magento2/blob/master/Setup/UpgradeSchema.php#L47 so we could use it. You need to do a similar thing.